### PR TITLE
Quick Start for Existing Users V2 - Changes to Existing Tasks - Reader (Part ii)

### PR DESCRIPTION
**Pull request by ashiagr**
Parent #16349

This PR makes following main changes for the `FOLLOW_SITE` task on the `Reader` tab:

1. Moves existing QS logic from reader post list to parent reader fragment and vm in 6b6264e (no. of lines have increased due to this). 
    #### [This is done as `Discover` tab fragment `ReaderDiscoverFragment`  is separate to other tab fragments (which are instances of `ReaderPostListFragment`). Both `DiscoverFragment`  and `ReaderPostListFragment` are created from `ReaderFragment` and it is easier to manage QS steps from this fragment + vm when we introduce `Discover` tab step which involves observing tab changes in the `ReaderViewModel`.]
2. Displays QS focus point for `Settings` menu. Note that previously QS focus point was not shown for `Search` menu. The task is marked complete when `Settings` menu is tapped.

3. ~~Adds intermediate `Discover` tab step if not on `Discover` tab.~~ Auto-selects`Discover` tab step if not on `Discover` tab.

https://user-images.githubusercontent.com/1405144/165953529-0977953b-0be2-4cde-9938-1079c97eceaa.mp4

To test:

1. Launch the app with a site having a quick start in progress.
2. Go to `My Site` tab.
3. Select `Grow your audience` from `Next Steps` card on `My Site` screen. The card will be present on the `Initial Screen `tab set in the `Settings`.
5. Click `Connect with other sites` row.
6. ✅ Notice that quick start flow for the task is updated as follows:
    
   Discover tab is auto-selected if not already selected.
7. ✅ Notice that quick start focus points are removed on navigating away from the `Reader` tab but retained on device rotation.
8. ✅ Notice that the task is marked complete only when `Settings` menu step of Reader Discover is completed. This can be checked by enabling `Collect information` in `Me` -> `App Settings `-> `Privacy settings` and checking for log:
 `🔵 Tracked: quick_start_task_completed, Properties: {"default_tab_experiment":<tab name>,"task_name":"follow_site"}`
or by running unit test with description: `given pending follow site qs task, when settings menu clicked, then setting menu step is completed`

Review Instructions:

Review tests collectively at the end (as opposed to commit-by-commit) as there were changes to the flow.

## Regression Notes
1. Potential unintended areas of impact
- Quick start focus points, snackbars are not updated based on whether the user was already on the Discover tab or not.
- Any crashes due to adding/ updating custom views for `Discover` tab and `Settings` menu.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually (see To Test: section) and added unit tests.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests in `ReaderViewModel` for updated quick start steps.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by osullivanchris — APPROVED** [review]
Hey @ashiagr everything looks good in the description and videos, as expected.

It behaved a bit funny for me when I installed the build
1. When I tapped on 'Reader' it gave me the settings task for a second before I got the discover task. Maybe it didn't realise what tab I was on
2. I then couldn't tap on discover to make the task move forward. 

I am going to take a lunch break now but I'll try it again later and see if I can reproduce.

I'm approving the overall approach though based on the task description etc and assume this is a bug we'll squash.

**Comment by osullivanchris** [comment]
@ashiagr I got another bug where the settings icon in reader didn't display at all. But the pointer still focused on where it should be. Are you seeing any of these issues or is it something on my end?

**Comment by ashiagr** [comment]
> I got another bug where the settings icon in reader didn't display at all. But the pointer still focused on where it should be. 

I think I know this issue. It should be an easy fix. I'm converting this PR to draft and might even close this one to update the flow based on Slack discussion to auto select Discover tab. Let's hold off further review and testing till then.

**Comment by osullivanchris** [comment]
> I  think I know this issue. It should be an easy fix. I'm converting this PR to draft and might even close this one to update the flow based on Slack discussion to auto select Discover tab. Let's hold off further review and testing till then.

Yeah sounds good! A lot of this logic might no longer apply.

**Comment by ashiagr** [comment]
👋 @osullivanchris 

Flows updated to 
- auto-select `Discover` tab (if not on discover) - ab46fbd
- show combined messages based on Settings availability - b66add8. This should also resolve the following bug: 
>  the settings icon in reader didn't display at all. But the pointer still focused on where it

https://user-images.githubusercontent.com/1405144/165953529-0977953b-0be2-4cde-9938-1079c97eceaa.mp4

I could not reproduce this bug:
> I then couldn't tap on discover to make the task move forward.

Let me know if you can reproduce it. 

Ready for another look. 🙇‍♀️

**Comment by osullivanchris** [comment]
Hey @ashiagr 👋 

> auto-select Discover tab (if not on discover)

this worked as expected. I was directed to Discover.

> I could not reproduce this bug: "I then couldn't tap on discover to make the task move forward."

That bug was happening in the two step flow, so its no longer relevant. Its not happening now.

> the settings icon in reader didn't display at all. But the pointer still focused on where it

however I'm still seeing this one ☝️  Here's a video 👇 As well as the settings icon missing, I also feel like the snackbar dismisses a bit quickly. Probably because the content took a couple seconds to load, and the snackbar was gone by the time the page fully loaded.


https://user-images.githubusercontent.com/50576199/165978532-9491ffc4-6552-4e35-aad1-607472703bd7.mp4

**Comment by ashiagr** [comment]
👋  @osullivanchris 

> the settings icon missing

Thank for the video. Fixed icon tint color in 4f05199.

> I also feel like the snackbar dismisses a bit quickly. 

Increased snackbar duration in 66e6ab6.

**Comment by osullivanchris** [comment]
@ashiagr sounds good, thanks!

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/quicklinksribbon/QuickLinkRibbonItemViewHolder.kt
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/quicklinksribbon/QuickLinkRibbonItemViewHolder.kt]: Nitpick: Nice Improvement 👍🏼. How about we rename it to `quickLinksRibbonItemQuickStartFocusPoint`?. 🤔 . Reference [comment](https://github.com/wordpress-mobile/WordPress-Android/pull/16361#discussion_r853730739)

**Review by AjeshRPai — APPROVED** [review]
I have looked at the code changes and tested them as per the instructions. Everything looks good to me. Great Job @ashiagr. 👍 🙌 . I left a nitpick for you to address.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/quicklinksribbon/QuickLinkRibbonItemViewHolder.kt
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/quicklinksribbon/QuickLinkRibbonItemViewHolder.kt]: Removed `Ribbon` and renamed to `quickLinkItemQuickStartFocusPoint` to match with `quickLinkItem` naming convention in the same class. Without `Ribbon` in the name, I think it is fine to keep `quickLink` singular.

In 7314eb4

**Comment by ashiagr** [comment]
Hey @jostnes 

I'm merging this PR as I need it for the next task and it has two approvals. But feel free to comment if you find any issues.

Thank you!

**Pull request merged**
